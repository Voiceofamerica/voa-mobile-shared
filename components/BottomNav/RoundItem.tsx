
import * as React from 'react'
import { Subscription } from 'rxjs/Subscription'

import { roundItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
}

function RoundItem ({ children, onClick = () => null }: Props) {
  return (
    <button className={roundItem} onClick={() => onClick()}>
      {
        children
      }
    </button>
  )
}

export default RoundItem
