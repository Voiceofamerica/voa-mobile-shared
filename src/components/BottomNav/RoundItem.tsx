
import * as React from 'react'

import { roundItem, activeItem } from './BottomNav.scss'

export type Props = React.Props<any> & {
  onClick?: () => void,
  active?: boolean,
  style?: React.CSSProperties,
  className?: string,
}

function RoundItem ({ className = '', style, children, onClick = () => null, active = false }: Props) {
  const fullClassName = active
    ? `${roundItem} ${activeItem} ${className}`
    : `${roundItem} ${className}`
  return (
    <div className={fullClassName} style={style} onClick={() => onClick()}>
      {
        children
      }
    </div>
  )
}

export default RoundItem
