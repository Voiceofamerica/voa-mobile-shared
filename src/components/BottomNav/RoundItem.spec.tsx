
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import RoundItem from './RoundItem'

describe(`<${RoundItem.name} />`, () => {
  describe('shapshots', () => {
    it('should render empty', () => {
      const element = create((
        <RoundItem />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given children', () => {
      const element = create((
        <RoundItem>
          <div>child1</div>
          <div>child2</div>
        </RoundItem>
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render active if prop set to true', () => {
      const element = create((
        <RoundItem active />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given style', () => {
      const element = create((
        <RoundItem style={{ top: 0, left: 0, right: 0, bottom: 0 }} />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given className', () => {
      const element = create((
        <RoundItem className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onClick', () => {
    it('should be called whenever the element is clicked', () => {
      const onClick = jest.fn()

      const item = shallow((
        <RoundItem onClick={onClick} />
      ))

      item.simulate('click')

      expect(onClick).toHaveBeenCalledTimes(1)
    })

    it('should not throw an error if not provided and the element is clicked', () => {
      const item = shallow((
        <RoundItem />
      ))

      item.simulate('click')
    })
  })
})
