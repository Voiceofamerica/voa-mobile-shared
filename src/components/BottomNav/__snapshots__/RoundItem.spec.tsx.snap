// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundItem /> shapshots should render active if prop set to true 1`] = `
<button
  className="roundItem activeItem "
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<RoundItem /> shapshots should render empty 1`] = `
<button
  className="roundItem "
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<RoundItem /> shapshots should render with the given children 1`] = `
<button
  className="roundItem "
  onClick={[Function]}
  style={undefined}
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</button>
`;

exports[`<RoundItem /> shapshots should render with the given className 1`] = `
<button
  className="roundItem myClass"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<RoundItem /> shapshots should render with the given style 1`] = `
<button
  className="roundItem "
  onClick={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
/>
`;
